lib/src/map/camera/camera.dart

```diff
@@ -249,87 +248,89 @@ class MapCamera {
-  Bounds<double> get pixelBounds =>
+  Rect get pixelBounds =>
       _pixelBounds ?? (_pixelBounds = pixelBoundsAtZoom(zoom));
 
   /// Calculates the pixel bounds of this [MapCamera] at the given [zoom].
-  Bounds<double> pixelBoundsAtZoom(double zoom) {
-    var halfSize = size / 2;
+  Rect pixelBoundsAtZoom(double zoom) {
+    Size cameraSize = size;
     if (zoom != this.zoom) {
       final scale = getZoomScale(this.zoom, zoom);
-      halfSize = size / (scale * 2);
+      cameraSize = size / (scale * 2);
     }
-    final pixelCenter = project(center, zoom).floor().toDoublePoint();
-    return Bounds(pixelCenter - halfSize, pixelCenter + halfSize);
+    final pixelCenter = project(center, zoom).floor();
+
+    return Rect.fromCenter(
+        center: pixelCenter,
+        width: cameraSize.width,
+        height: cameraSize.height);
   }
 
   /// This will convert a latLng to a position that we could use with a widget
   /// outside of FlutterMap layer space. Eg using a Positioned Widget.
-  Point<double> latLngToScreenPoint(LatLng latLng) {
-    final nonRotatedPixelOrigin = project(center, zoom) - nonRotatedSize / 2.0;
+  Offset latLngToScreenOffset(LatLng latLng) {
+    final nonRotatedPixelOrigin =
+        project(center, zoom) - nonRotatedSize.center(Offset.zero);
 
-    var point = crs.latLngToPoint(latLng, zoom);
+    var point = crs.latLngToOffset(latLng, zoom);
 
-    final mapCenter = crs.latLngToPoint(center, zoom);
+    final mapCenter = crs.latLngToOffset(center, zoom);
 
     if (rotation != 0.0) {
-      point = rotatePoint(mapCenter, point, counterRotation: false);
+      point = rotateOffset(mapCenter, point, counterRotation: false);
     }
 
     return point - nonRotatedPixelOrigin;
   }
 
   /// Calculate the [LatLng] coordinates for a [localPoint].
-  LatLng pointToLatLng(Point localPoint) {
-    final localPointCenterDistance = Point(
-      (nonRotatedSize.x / 2) - localPoint.x,
-      (nonRotatedSize.y / 2) - localPoint.y,
+  LatLng offsetToLatLng(Offset localPoint) {
+    final localPointCenterDistance = Offset(
+      (nonRotatedSize.width / 2) - localPoint.dx,
+      (nonRotatedSize.height / 2) - localPoint.dy,
     );
-    final mapCenter = crs.latLngToPoint(center, zoom);
+    final mapCenter = crs.latLngToOffset(center, zoom);
 
     var point = mapCenter - localPointCenterDistance;
 
     if (rotation != 0.0) {
-      point = rotatePoint(mapCenter, point);
+      point = rotateOffset(mapCenter, point);
     }
 
-    return crs.pointToLatLng(point, zoom);
+    return crs.offsetToLatLng(point, zoom);
   }
 
   /// Sometimes we need to make allowances that a rotation already exists, so
   /// it needs to be reversed (pointToLatLng), and sometimes we want to use
   /// the same rotation to create a new position (latLngToScreenpoint).
   /// counterRotation just makes allowances this for this.
-  Point<double> rotatePoint(
-    Point<double> mapCenter,
-    Point<double> point, {
+  Offset rotateOffset(
```

Comment:
Whilst we use 'point' for variable and parameter names, I would suggest keeping this as `rotatePoint`, but I'm not sure (it would be inconsistent with other external methods). What do you think?

Reply:
agreed!